src/pyOpenMS/pxds/IDFilter.pxd

```diff
@@ -18,83 +18,166 @@ from ChromatogramPeak cimport *
 cdef extern from "<OpenMS/FILTERING/ID/IDFilter.h>" namespace "OpenMS":
 
     cdef cppclass IDFilter:
-
-        IDFilter() nogil except +
-        IDFilter(IDFilter &) nogil except + # compiler
-
-        Size countHits(libcpp_vector[PeptideIdentification] identifications) nogil except +
-        Size countHits(libcpp_vector[ProteinIdentification] identifications) nogil except +
+         # wrap-doc:
+                #   Finds the best-scoring hit in a vector of peptide or protein identifications
+                #   -----
+                #   This class provides functions for filtering collections of peptide or protein identifications according to various criteria.
+                #   It also contains helper functions and classes (functors that implement predicates) that are used in this context.
+                #   -----
+                #   The filter functions modify their inputs, rather than creating filtered copies.
+                #   -----
+                #   Most filters work on the hit level, i.e. they remove peptide or protein hits from peptide or protein identifications (IDs).
+                #   A few filters work on the ID level instead, i.e. they remove peptide or protein IDs from vectors thereof.
+                #   Independent of this, the inputs for all filter functions are vectors of IDs, because the data most often comes in this form.
+                #   This design also allows many helper objects to be set up only once per vector, rather than once per ID.
+                #   -----
+                #   The filter functions for vectors of peptide/protein IDs do not include clean-up steps (e.g. removal of IDs without hits, reassignment of hit ranks, ...).
+                #   They only carry out their specific filtering operations.
+                #   This is so filters can be chained without having to repeat clean-up operations.
+                #   The group of clean-up functions provides helpers that are useful to ensure data integrity after filters have been applied, but it is up to the individual developer to use them when necessary.
+                #   -----
+                #   The filter functions for MS/MS experiments do include clean-up steps, because they filter peptide and protein IDs in conjunction and potential contradictions between the two must be eliminated.
+
+        IDFilter()           nogil except + 
```

Comment:
```suggestion
        IDFilter() nogil except + 
```

Reply:
FYI: You do not have to use the commits, but also can change it locally and push.